walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -76,41 +92,45 @@
     List<TimeRange> endOrderedTimeRanges = getOrderedTimeRanges(occupiedTimeRanges,
                                                                 TimeRange.ORDER_BY_END);
 
-    // Potentially add the {@code TimeRange} before the first existing {@code TimeRange}.
+    // Potentially add the {@code TimeRange} before the first occupied {@code TimeRange}.
     TimeRange firstStartingTimeRange = startOrderedTimeRanges.get(0);
     checkDurationAndAddAvailableTimeRange(requestedDuration,
                                           TimeRange.START_OF_DAY, firstStartingTimeRange.start(),
                                           availableTimeRanges);
 
-    // Find available {@code TimeRange} by looking at the available time between one ending
-    // {@code TimeRange} and the immediately following starting {@code TimeRange}.
+    // Find available {@code TimeRange} by looking at the available time between the end of one
+    // occupied {@code TimeRange} and the start of the next occupied {@code TimeRange}:
+    //   |----ending {@code TimeRange}----| available time |----starting {@code TimeRange}----|
     ListIterator<TimeRange> startOrderedTimeRangesIterator = startOrderedTimeRanges.listIterator();
     ListIterator<TimeRange> endOrderedTimeRangesIterator = endOrderedTimeRanges.listIterator();
     TimeRange endingTimeRange = endOrderedTimeRangesIterator.next();
     TimeRange startingTimeRange = startOrderedTimeRangesIterator.next();
+
+    findGapsInOccupiedTimeRanges:
     while (true) {
-      // There is no available time after the current ending {@code TimeRange} and before the
-      // immediately following starting {@code TimeRange}.
-      if (endingTimeRange.overlaps(startingTimeRange)) {
+
+      while (endingTimeRange.overlaps(startingTimeRange)) {
         if (startOrderedTimeRangesIterator.hasNext()) {
           startingTimeRange = startOrderedTimeRangesIterator.next();
         } else {
-          break;
+          break findGapsInOccupiedTimeRanges;
         }
-        continue;
       }
-      // There is another {@code TimeRange} that ends later than the current ending
-      // {@code TimeRange} + before the immediately following, starting {@code TimeRange}.
-      // (Since there exists a {@code TimeRange} starting later than the current ending
-      //  {@code TimeRange}, there must be other {@code TimeRange} ending even later. Therefore,
-      //  the {@code endOrderedPointer} is guaranteed to not go out of range.)
+
+      // There is another occupied {@code TimeRange} between the current ending {@code TimeRange}
+      // and the next occupied {@code TimeRange}.
```

Comment:
Maybe instead, something like "Find the latest occupied timerange before startingtimerange".